src/core/proc_maps/linux_maps.rs

```diff
@@ -1,32 +1,33 @@
+use libc;
 use std;
 use std::fs::File;
 use std::io::Read;
-use libc::pid_t;
+
+pub type Pid = libc::pid_t;
+
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct MapRange {
-    pub range_start: usize,
-    pub range_end: usize,
-    pub offset: usize,
-    pub dev: String,
-    pub flags: String,
-    pub inode: usize,
-    pub pathname: Option<String>,
+    range_start: usize,
+    range_end: usize,
+    offset: usize,
+    dev: String,
+    flags: String,
+    inode: usize,
+    pathname: Option<String>,
 }
 
 impl MapRange {
-    pub fn contains_addr(&self, addr: usize) -> bool {
-        addr >= self.range_start && addr <= self.range_end
-    }
-}
-
-pub fn maps_contain_addr(addr: usize, maps: &Vec<MapRange>) -> bool {
-    maps.iter().any({ |map| map.contains_addr(addr) })
+    pub fn size(&self) -> usize { self.range_end - self.range_start }
+    pub fn start(&self) -> usize { self.range_start }
+    pub fn filename(&self) -> &Option<String> { &self.pathname }
+    pub fn is_exec(&self) -> bool { &self.flags[2..3] == "x" }
+    pub fn is_write(&self) -> bool { &self.flags[1..2] == "w" }
+    pub fn is_read(&self) -> bool { &self.flags[0..1] == "r" }
```

Comment:
it doesn't really matter but it might be nice (since there's sort of an implicit interface here) to implement a real interface.

Reply:
Good point - making the implied interface explicit seems like a great idea!

I’ve added here: https://github.com/rbspy/rbspy/pull/166/commits/bfe6ad2e8e0f54fe64d9f39f9d6faa4d3cde73a9 but got a little hung up on naming the trait. I went with IMapRange as a placeholder (so as not to conflict with the MapRange structs) but I am totally open to better suggestions.